### PR TITLE
chore(experiment): Adding target details inside chaosresult for the experiments which doesn't contain helper pod

### PR DESCRIPTION
**Pull request by ispeakc0de**
Signed-off-by: shubhamchaudhary <shubham@chaosnative.com>

- Adding target details inside chaosresult for the experiments which doesn't contain helper pod
- It will add three attributes:
    - name: name of pod/node
    - kind: kind of application i.e pod,node
    - chaosStatus: status of chaos[injected, reverted, targeted]

Tested for:
- ebs-loss :heavy_check_mark: 
- ec2-terminate-by-id :heavy_check_mark: 
- ec2-terminate-by-tag  :heavy_check_mark: 
- kafka-pod-delete  :heavy_check_mark: 
- node-drain  :heavy_check_mark: 
- node-taint :heavy_check_mark: 
- pod-cpu-hog :heavy_check_mark: 
- pod-memory-hog  :heavy_check_mark: 
- kubelet-service-kill  :heavy_check_mark: 
- node-cpu-hog :heavy_check_mark: 
- node-memory-hog  :heavy_check_mark: 
- node-io-stress  :heavy_check_mark: 
- node-restart :heavy_check_mark: 
- pumba lib :heavy_check_mark:

### CODE REVIEW

**Comment by ispeakc0de** [comment]
### pod-cpu-hog
```yaml
Name:         helloservice-pod-delete1-pod-cpu-hog
Namespace:    shubham
Labels:       name=helloservice-pod-delete1-pod-cpu-hog
Annotations:  <none>
API Version:  litmuschaos.io/v1alpha1
Kind:         ChaosResult
Metadata:
  Creation Timestamp:  2021-04-28T19:21:06Z
  Generation:          5
  Resource Version:    346241
  Self Link:           /apis/litmuschaos.io/v1alpha1/namespaces/shubham/chaosresults/helloservice-pod-delete1-pod-cpu-hog
  UID:                 f88bb56a-f170-44c8-a9a5-b37945d2d0df
Spec:
  Engine:      helloservice-pod-delete1
  Experiment:  pod-cpu-hog
Status:
  Experiment Status:
    Fail Step:                 Chaos injection stopped!
    Phase:                     Completed
    Probe Success Percentage:  0
    Verdict:                   Stopped
  History:
    Failed Runs:   0
    Passed Runs:   1
    Stopped Runs:  2
    Targets:
      Chaos Status:  recovered
      Kind:          pod
      Target:        nginx1-78cb94bbd5-qz5xm
      Chaos Status:  recovered
      Kind:          pod
      Target:        nginx1-78cb94bbd5-7g4bc
Events:
  Type     Reason   Age    From                      Message
  ----     ------   ----   ----                      -------
  Normal   Awaited  5m38s  pod-cpu-hog-xej9d5-q2fs7  experiment: pod-cpu-hog, Result: Awaited
  Normal   Pass     4m29s  pod-cpu-hog-xej9d5-q2fs7  experiment: pod-cpu-hog, Result: Pass
  Normal   Awaited  41s    pod-cpu-hog-wpsbmv-g9fjh  experiment: pod-cpu-hog, Result: Awaited
  Warning  Summary  14s    pod-cpu-hog-wpsbmv-g9fjh  pod-cpu-hog experiment has been aborted
```

**Comment by ispeakc0de** [comment]
### node taint
```yaml
Name:         helloservice-pod-delete1-node-taint
Namespace:    shubham
Labels:       app.kubernetes.io/part-of=litmus
              chaosUID=d175f365-ea58-4252-a9c5-4f4111241b99
              controller-uid=ae006fdb-27c9-4dad-ad13-eaed4cf54df0
              job-name=node-taint-8dcwui
              name=helloservice-pod-delete1-node-taint
Annotations:  <none>
API Version:  litmuschaos.io/v1alpha1
Kind:         ChaosResult
Metadata:
  Creation Timestamp:  2021-04-28T19:29:50Z
  Generation:          2
  Resource Version:    346959
  Self Link:           /apis/litmuschaos.io/v1alpha1/namespaces/shubham/chaosresults/helloservice-pod-delete1-node-taint
  UID:                 f40cc911-7661-48e3-854b-e511b4967a55
Spec:
  Engine:      helloservice-pod-delete1
  Experiment:  node-taint
Status:
  Experiment Status:
    Fail Step:                 N/A
    Phase:                     Completed
    Probe Success Percentage:  100
    Verdict:                   Pass
  History:
    Failed Runs:   0
    Passed Runs:   1
    Stopped Runs:  0
    Targets:
      Chaos Status:  recovered
      Kind:          node
      Target:        ip-192-168-1-160.ec2.internal
Events:
  Type    Reason   Age   From                     Message
  ----    ------   ----  ----                     -------
  Normal  Awaited  59s   node-taint-8dcwui-rzbrw  experiment: node-taint, Result: Awaited
  Normal  Pass     28s   node-taint-8dcwui-rzbrw  experiment: node-taint, Result: Pass
```

**Comment by ispeakc0de** [comment]
## pumba network-loss
```yaml
Name:         helloservice-pod-delete1-pod-network-loss
Namespace:    shubham
Labels:       name=helloservice-pod-delete1-pod-network-loss
Annotations:  <none>
API Version:  litmuschaos.io/v1alpha1
Kind:         ChaosResult
Metadata:
  Creation Timestamp:  2021-04-28T20:16:41Z
  Generation:          4
  Resource Version:    355891
  Self Link:           /apis/litmuschaos.io/v1alpha1/namespaces/shubham/chaosresults/helloservice-pod-delete1-pod-network-loss
  UID:                 6d8d9821-ed20-4fa1-9005-78febb178107
Spec:
  Engine:      helloservice-pod-delete1
  Experiment:  pod-network-loss
Status:
  Experiment Status:
    Fail Step:                 Chaos injection stopped!
    Phase:                     Completed
    Probe Success Percentage:  0
    Verdict:                   Stopped
  History:
    Failed Runs:   0
    Passed Runs:   1
    Stopped Runs:  1
    Targets:
      Chaos Status:  targeted
      Kind:          pod
      Target:        nginx1-78cb94bbd5-7g4bc
Events:
  Type     Reason   Age    From                           Message
  ----     ------   ----   ----                           -------
  Normal   Awaited  2m27s  pod-network-loss-mdpm3s-pdrlw  experiment: pod-network-loss, Result: Awaited
  Normal   Pass     79s    pod-network-loss-mdpm3s-pdrlw  experiment: pod-network-loss, Result: Pass
  Normal   Awaited  44s    pod-network-loss-u48dna-98grm  experiment: pod-network-loss, Result: Awaited
  Warning  Summary  26s    pod-network-loss-u48dna-98grm  pod-network-loss experiment has been aborted

```

File: chaoslib/litmus/ebs-loss/lib/ebs-loss.go [hunk 1]
```diff
@@ -112,6 +112,8 @@ func EBSVolumeDetach(experimentsDetails *experimentTypes.ExperimentDetails) erro
 		}
 	}
 
+	common.SetTargets(experimentsDetails.EBSVolumeID, "injected", "EBS Volume ID", chaosDetails)
```
Comment (uditgaurav) [inline chaoslib/litmus/ebs-loss/lib/ebs-loss.go:115]:
```suggestion
	common.SetTargets(experimentsDetails.EBSVolumeID, "injected", "EBS", chaosDetails)
```

File: chaoslib/litmus/ebs-loss/lib/ebs-loss.go [hunk 2]
```diff
@@ -153,6 +155,8 @@ func EBSVolumeAttach(experimentsDetails *experimentTypes.ExperimentDetails) erro
 		}
 	}
 
+	common.SetTargets(experimentsDetails.EBSVolumeID, "reverted", "EBS Volume ID", chaosDetails)
```
Comment (uditgaurav) [inline chaoslib/litmus/ebs-loss/lib/ebs-loss.go:158]:
```suggestion
	common.SetTargets(experimentsDetails.EBSVolumeID, "reverted", "EBS", chaosDetails)
```

File: chaoslib/litmus/ec2-terminate-by-id/lib/ec2-terminate-by-id.go [hunk 3]
```diff
@@ -75,6 +75,8 @@ loop:
 				return errors.Errorf("ec2 instance failed to stop, err: %v", err)
 			}
 
+			common.SetTargets(id, "injected", "EC2 Instance ID", chaosDetails)
```
Comment (uditgaurav) [inline chaoslib/litmus/ec2-terminate-by-id/lib/ec2-terminate-by-id.go:78]:
```suggestion
			common.SetTargets(id, "injected", "EC2", chaosDetails)
```
Please check at other places as well.

File: chaoslib/litmus/ec2-terminate-by-id/lib/ec2-terminate-by-id.go [hunk 4]
```diff
@@ -150,6 +154,7 @@ loop:
 			if err != nil {
 				return errors.Errorf("ec2 instance failed to stop, err: %v", err)
 			}
+			common.SetTargets(id, "injected", "EC2 Instance ID", chaosDetails)
```
Comment (uditgaurav) [inline chaoslib/litmus/ec2-terminate-by-id/lib/ec2-terminate-by-id.go:157]: How about putting this `SetTargets` inside `EC2Stop` and `EC2Start` functions that will be common to all?

File: chaoslib/litmus/ec2-terminate-by-id/lib/ec2-terminate-by-id.go [hunk 5]
```diff
@@ -180,6 +185,7 @@ loop:
 				if err != nil {
 					return errors.Errorf("ec2 instance failed to start, err: %v", err)
 				}
+				common.SetTargets(id, "reverted", "EC2 Instance ID", chaosDetails)
```
Comment (uditgaurav) [inline chaoslib/litmus/ec2-terminate-by-id/lib/ec2-terminate-by-id.go:188]: One consideration here is in the case of Managed Nodegroup the EC2 will not start so how are we confirming the revert?

File: chaoslib/litmus/node-memory-hog/lib/node-memory-hog.go [hunk 6]
```diff
@@ -273,7 +281,7 @@ func CalculateMemoryConsumption(experimentsDetails *experimentTypes.ExperimentDe
 
 	if experimentsDetails.MemoryConsumptionMebibytes == 0 {
 		if experimentsDetails.MemoryConsumptionPercentage == 0 {
-			log.Info("Neither of MemoryConsumptionPercentage or MemoryConsumptionMebibytes provided, proceeding with a default MemoryConsumptionPercentage value of 30%")
+			log.Info("Neither of MemoryConsumptionPercentage or MemoryConsumptionMebibytes provided, proceeding with a default MemoryConsumptionPercentage value of 30%%")
```
Comment (uditgaurav) [inline chaoslib/litmus/node-memory-hog/lib/node-memory-hog.go:284]:
```suggestion
			log.Info("Neither of MemoryConsumptionPercentage or MemoryConsumptionMebibytes provided, proceeding with a default MemoryConsumptionPercentage value of 30%")
```

File: chaoslib/litmus/ec2-terminate-by-id/lib/ec2-terminate-by-id.go (hunk 4 above)
Comment (ispeakc0de) [reply chaoslib/litmus/ec2-terminate-by-id/lib/ec2-terminate-by-id.go:157]: yes we add there, but we need to pass an extra argument there

File: chaoslib/litmus/ec2-terminate-by-id/lib/ec2-terminate-by-id.go (hunk 5 above)
Comment (ispeakc0de) [reply chaoslib/litmus/ec2-terminate-by-id/lib/ec2-terminate-by-id.go:188]: as managed node-groups will automatically revert it so we are considering it as true positive case and patching as revert

File: chaoslib/litmus/node-memory-hog/lib/node-memory-hog.go (hunk 6 above)
Comment (ispeakc0de) [reply chaoslib/litmus/node-memory-hog/lib/node-memory-hog.go:284]: in log, if we want to print a percentage symbol we need to put two percentage symbols. This is correct

**Comment by uditgaurav** [comment]
Looks Good. Needs a pipeline run!

**Review by ksatchit — APPROVED** [review]

**Pull request merged**